x/supplier/keeper/prune_supplier_service_config_history.go

```diff
@@ -64,7 +77,7 @@ func (k Keeper) EndBlockerPruneSupplierServiceConfigHistory(
 		// Update the supplier's service config history with the pruned list.
 		supplier.ServiceConfigHistory = retainedServiceConfigs
 
-		k.SetSupplier(ctx, supplier)
+		k.SetSupplier(ctx, *supplier)
```

Comment:
Two lines above this, we are modifying a field from the pointer.

I don't think it should be an issue, but it does raise a flag that we should potentially make a copy of the supplier at the very top.

Avoid potential value/pointer concurrency issues.

Reply:
Will make the iterator return a value instead of a pointer. It will actually be closer to what `GetSupplier/GetAllSuppliers` return